File: src/org/cappuccino_project/ide/intellij/plugin/indices/ObjJSelectorInferredMethodIndex.java
package org.cappuccino_project.ide.intellij.plugin.indices;

import com.intellij.psi.stubs.StubIndexKey;
import org.cappuccino_project.ide.intellij.plugin.psi.ObjJSelectorLiteral;
import org.jetbrains.annotations.NotNull;

public class ObjJSelectorInferredMethodIndex extends ObjJMethodHeaderDeclarationsIndexBase<ObjJSelectorLiteral> {
    private static final ObjJSelectorInferredMethodIndex INSTANCE = new ObjJSelectorInferredMethodIndex();
    private static final StubIndexKey<String, ObjJSelectorLiteral> KEY = IndexKeyUtil.createIndexKey(ObjJSelectorInferredMethodIndex.class);
    private static final int VERSION = 0;

    public static ObjJSelectorInferredMethodIndex getInstance() {
        return INSTANCE;
    }

    @NotNull
    @Override
    public StubIndexKey<String, ObjJSelectorLiteral> getKey() {
        return KEY;
    }

    public int getVersion() {
        return super.getVersion() + VERSION;
    }

    @NotNull
    @Override
    protected Class<ObjJSelectorLiteral> getIndexedElementClass() {
        return ObjJSelectorLiteral.class;
    }
}

File: src/org/cappuccino_project/ide/intellij/plugin/indices/ObjJSelectorInferredMethodIndex.kt
package org.cappuccino_project.ide.intellij.plugin.indices

import com.intellij.openapi.project.Project
import com.intellij.psi.search.GlobalSearchScope
import com.intellij.psi.stubs.StringStubIndexExtension
import com.intellij.psi.stubs.StubIndex
import com.intellij.psi.stubs.StubIndexKey
import org.cappuccino_project.ide.intellij.plugin.psi.ObjJSelectorLiteral
import org.cappuccino_project.ide.intellij.plugin.psi.interfaces.ObjJMethodHeaderDeclaration
import org.cappuccino_project.ide.intellij.plugin.stubs.ObjJStubVersions

import java.util.ArrayList

class ObjJSelectorInferredMethodIndex : ObjJMethodHeaderDeclarationsIndexBase<ObjJSelectorLiteral>() {

    protected override val indexedElementClass: Class<ObjJSelectorLiteral>
        get() = ObjJSelectorLiteral::class.java

    override fun getKey(): StubIndexKey<String, ObjJSelectorLiteral> {
        return KEY
    }

    override fun getVersion(): Int {
        return super.getVersion() + VERSION
    }

    companion object {
        val instance = ObjJSelectorInferredMethodIndex()
        private val KEY = IndexKeyUtil.createIndexKey(ObjJSelectorInferredMethodIndex::class.java)
        private val VERSION = 1
    }
}
